data/src/main/java/me/androidbox/data/remote/authentication/AuthenticationRepositoryImp.kt

```diff
@@ -57,4 +57,17 @@ class AuthenticationRepositoryImp @Inject constructor(
             ResponseState.Failure((result.exceptionOrNull() ?: Exception()) as Exception)
         }
     }
+
+    override suspend fun authenticateUser(): ResponseState<Unit> {
+        val result = checkResult<Unit> {
+            authenticationService.authenticate()
+        }
+
+        return if(result.isSuccess) {
+            ResponseState.Success(Unit)
+        }
+        else {
+            ResponseState.Failure(result.exceptionOrNull() as? Exception ?: Exception())
```

Comment:
The as? Exception ?: Exception casting is weird

You could probably do:

```
val exception = result.exceptionOrNull() ?: Exception()
ResponseState.Failure(exception)

```

Reply:
@MKiperszmid The reason for the cast as exceptionOrNull return a Throwable and the Failure(...) only excepts a Exception. So I had to cast it to Throwable. 

However,  I have update the responseState to except a Throwable i.e.